tmt/steps/prepare/feature.py

```diff
@@ -0,0 +1,794 @@
+            return Distro.CENTOS_7
+
+        if os_release.get('NAME', '').startswith('CentOS Stream'):
+            if re.match(r'^8$|^8\.', os_release.get('VERSION_ID', '')):
+                return Distro.CENTOS_STREAM_8
+            if re.match(r'^9$|^9\.', os_release.get('VERSION_ID', '')):
+                return Distro.CENTOS_STREAM_9
+
+        return None
+
+    def get_guest_distro_name(self) -> Optional[str]:
+        """ Get guest distro name by parsing the guest facts """
+        os_release = self.guest.facts.os_release_content
+        if os_release is None:
+            return None
+        return os_release.get('PRETTY_NAME', None)
+
+
+FEDORA_EPEL_PACKAGE = 'epel-release'
+FEDORA_EPEL_NEXT_PACKAGE = 'epel-next-release'
+YUM_UTILS_PACKAGE = 'yum-utils'
+RHEL_7_EPEL_PACKAGE = 'https://dl.fedoraproject.org/pub/epel/epel-release-latest-7.noarch.rpm'
+RHEL_8_EPEL_PACKAGE = 'https://dl.fedoraproject.org/pub/epel/epel-release-latest-8.noarch.rpm'
+RHEL_9_EPEL_PACKAGE = 'https://dl.fedoraproject.org/pub/epel/epel-release-latest-9.noarch.rpm'
+RHEL_8_EPEL_NEXT_PACKAGE = \
+    'https://dl.fedoraproject.org/pub/epel/epel-next-release-latest-8.noarch.rpm'
+RHEL_9_EPEL_NEXT_PACKAGE = \
+    'https://dl.fedoraproject.org/pub/epel/epel-next-release-latest-9.noarch.rpm'
+
+# Repos of epel and epel-next. For details, please refer to files in the following:
+# 1) /etc/yum.repos.d/epel.repo
+# 2) /etc/yum.repos.d/epel-next.repo
+EPEL_REPOS = ['epel', 'epel-debuginfo', 'epel-source']
+EPEL_NEXT_REPOS = ['epel-next', 'epel-next-debuginfo', 'epel-next-source']
+
+
+class GuestConfigManager(enum.Enum):
+    DNFCM = 'dnf config-manager'
+    YUMCM = 'yum-config-manager'
+
+
+class ToggleableFeature(Feature):
+    def enable(self) -> None:
+        raise NotImplementedError
+
+    def disable(self) -> None:
+        raise NotImplementedError
+
+    def _execute(
+            self,
+            command: Command) -> Optional[tmt.utils.CommandOutput]:
+        """ Run a command on the guest. """
+
+        try:
+            return self.guest.execute(command, silent=True)
+
+        except tmt.utils.RunError as exc:
+            if exc.stdout and 'Please login as the user' in exc.stdout:
+                raise tmt.utils.GeneralError(f'Login to the guest failed.\n{exc.stdout}') from exc
+
+        return None
+
+    def _probe(self,
+               probes: List[Tuple[Command, GuestConfigManager]]) -> Optional[GuestConfigManager]:
+        """ Find a first successfull command. """
+
+        for command, outcome in probes:
+            if self._execute(self.guest, command):
+                return outcome
+
+        return None
+
+    def query_config_manager(self) -> Optional[GuestConfigManager]:
+        return self._probe([
+                (Command('dnf config-manager', '--version'), GuestConfigManager.DNFCM),
+                (Command('yum-config-manager', '--version'), GuestConfigManager.YUMCM)
+                ])
```

Comment:
It might be better to put the config manager probing into guest facts directly, there seems to be a duplication of necessary functionality.

Reply:
Let's skip this because of  `ansible` solution :-)